src/shell/shell.c

```diff
@@ -495,6 +495,84 @@ int flux_shell_rank_info_unpack (flux_shell_t *shell,
     return rc;
 }
 
+static json_t *flux_shell_get_jobspec_info_object (flux_shell_t *shell)
+{
+    json_t *o = NULL;
+    struct jobspec *jobspec;
+    if (!(jobspec = shell->info->jobspec))
+        return NULL;
+
+    if ((o = flux_shell_aux_get (shell, "shell::jobspec_info")))
+        return o;
+
+    /*  Only v1 supported for now:
+     */
+    if (jobspec->version == 1) {
+        o = json_pack ("{s:i s:i s:i s:i s:i s:i}",
+                       "version", jobspec->version,
+                       "ntasks", jobspec->task_count,
+                       "nslots", jobspec->slot_count,
+                       "cores_per_slot", jobspec->cores_per_slot,
+                       "nnodes", jobspec->node_count,
+                       "slots_per_node", jobspec->slots_per_node);
+    }
+    else
+        o = json_pack ("{s:i}", "version", jobspec->version);
+
+    if (o == NULL)
+        return NULL;
+
+    if (flux_shell_aux_set (shell,
+                            "shell::jobspec_info",
+                            o,
+                            (flux_free_f) json_decref) < 0) {
+        json_decref (o);
+        return NULL;
+    }
+    return o;
+}
+
+int flux_shell_get_jobspec_info (flux_shell_t *shell, char **json_str)
+{
+    json_t *o;
+    if (!shell || !json_str) {
+        errno = EINVAL;
+        return -1;
+    }
+    if (!(o = flux_shell_get_jobspec_info_object (shell)))
+        return -1;
+    *json_str = json_dumps (o, JSON_COMPACT);
```

Comment:
sets `json_str` on error.  I think we decided that was a pattern to avoid in flux?